lib/karafka/connection/messages_processor.rb

```diff
@@ -38,34 +38,22 @@ def process(group_id, kafka_messages)
         # @param controller [Karafka::BaseController] base controller descendant
         # @param mapped_topic [String] mapped topic name
         # @param kafka_messages [Array<Kafka::FetchedMessage>] raw messages from kafka
-        def process_batch(controller, mapped_topic, kafka_messages)
-          messages_batch = kafka_messages.map do |kafka_message|
-            # Since we support topic mapping (for Kafka providers that require namespaces)
-            # we have to overwrite topic with our mapped topic version
-            # @note For the default mapper, it will be the same as topic
-            # @note We have to use instance_variable_set, as the Kafka::FetchedMessage does not
-            #   provide attribute writers
-            kafka_message.instance_variable_set(:'@topic', mapped_topic)
```

Comment:
The remapping was causing the message to be consumed in an endless loop in my testing as ruby-kafka was seeing the incorrect (non-actual) topic name. I think the message itself should always have the real topic name in my opinion and the karafka code interpreting it with the mapper as needed... perhaps I misunderstand why the message must be altered at this level?